### PR TITLE
Add webdav http extension methods

### PR DESCRIPTION
**Pull request by beshrkayali**
WebDAV (as defined in [rfc4918](https://datatracker.ietf.org/doc/html/rfc4918)) extends the set of standard HTTP methods allowed. Custom methods were allowed before 1.5 when `request` function's `httpMethod` argument was of type `string` (which was deprecated in favor of `HttpMethod` enum type in #15919).

### CODE REVIEW

File: lib/pure/httpcore.nim [hunk 1]
```diff
@@ -50,6 +50,18 @@ type
...
+    HttpLock = "LOCK"            ## put a lock on a resource
+    HttpMkCol = "MKCOL"          ## create collections
+    HttpMove = "MOVE"            ## move a resource from one URI to another
+    HttpPropFind = "PROPFIND"    ## retrieve properties, stored as XML,
+                                 ## from a web resource. It is also overloaded
+                                 ## to allow one to retrieve the collection
+                                 ## structure
+    HttpPropPatch = "PROPPATCH"  ## change and delete multiple properties on a
+                                 ## resource in a single atomic act
+    HttpUnlock = "UNLOCK"        ## remove a lock from a resource
```
Comment (konsumlamm) [inline lib/pure/httpcore.nim]:
```suggestion
    HttpCopy = "COPY"            ## Copies a resource from one URI to another.
    HttpLock = "LOCK"            ## Puts a lock on a resource.
    HttpMkCol = "MKCOL"          ## Creates collections.
    HttpMove = "MOVE"            ## Moves a resource from one URI to another.
    HttpPropFind = "PROPFIND"    ## Retrieve properties, stored as XML,
                                 ## from a web resource. It is also overloaded
                                 ## to allow one to retrieve the collection
                                 ## structure.
    HttpPropPatch = "PROPPATCH"  ## Changes and deletes multiple properties on a
                                 ## resource in a single atomic operation.
    HttpUnlock = "UNLOCK"        ## Removes a lock from a resource.        
```

**Review by Varriount — APPROVED** [review]

**Comment by Varriount** [comment]
@Araq How should this be handled? As-is, the change breaks backwards compatibility. What I can think of is either modifying the relevant procedures to once-again accept strings, or break the extension methods into their own enum.

**Comment by Araq** [comment]
I don't know. It's ok to break it for version 2.0 but the better option might be an overload that accepts a `distinct string` and to use constants instead of enum fields.

**Comment by dom96** [comment]
>  It's ok to break it for version 2.0 but the better option might be an overload that accepts a distinct string and to use constants instead of enum fields.

or just provide a converter between the enum and the `distinct string`, I think that would work very well

**Review by Varriount — APPROVED** [review]

**Pull request closed without merging**